tests/testsuite/namespace.go

```diff
@@ -204,10 +205,20 @@ func removeNamespaces() {
 		fmt.Printf("Waiting for namespace %s to be removed, this can take a while ...\n", namespace)
 		EventuallyWithOffset(1, func() error {
 			return virtCli.CoreV1().Namespaces().Delete(context.Background(), namespace, metav1.DeleteOptions{})
-		}, 240*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), fmt.Sprintf("should successfully delete namespace '%s'", namespace))
+		}, 240*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), fmt.Sprintf("should successfully delete namespace: \n%s", dumpNamespaceObject(namespace)))
```

Comment:
maybe it triggers the functions when it builds the tests skeleton ?
(and this cause the failures)

Reply:
Do you think this change is the cause of the ci failing? It doesn't look like it I think

Reply:
well there is a small chance that the ns is gone in the time between the eventually failed and the dump is started.. I added another if to address that. if the tests failed on that then it should be clearer now

Reply:
That can happen of course (https://github.com/kubevirt/kubevirt/pull/8021#discussion_r912443011)
but i think something else happened here (i.e the function was running even before the condition ?)
anyhow, not sure about it, and the correct code is fine, with little note that in case we do see the new print of the aborting (then it means that after the timeout, it did finish to clean the ns)

Reply:
>with little note that in case we do see the new print of the aborting (then it means that after the timeout, it did finish to clean the ns

I agree, but that's also interesting to learn. Let's see

Reply:
Can you please inject an error like finalizer exists so we see what happens ?
(If indeed the function was running before the condition, lets see that the output is good enough, but it means that if anything changed in between, it might not be reflected)

(btw the reason i suspect it is that before the latest fix, we saw the error that the NS is not found, at the end of all the tests, after the NS do exists because it was used for the tests, without the error of the Expect line)

Reply:
Do you mean to print if any object under this ns has a finalizer hanging? If so - then the namespace already prints that (in fact, that is partly what this PR is meant for):
```
        {
          "type": "NamespaceFinalizersRemaining",
          "status": "True",
          "lastTransitionTime": "2022-06-29T11:28:03Z",
          "reason": "SomeFinalizersRemain",
          "message": "Some content in the namespace has finalizers remaining: ram-sam-sam in 2 resource instances"
        }
```

Reply:
I mean to inject an error please to see we are acting as expected
With little note that perhaps the function executing is done before the condition is checked, so it means that what we will see it the NS status before the attempt to delete it, which might changed some stuff
basically it is not perfect if so, but might be good enough